src/NuGet.Core/NuGet.Common/Migrations/MigrationRunner.cs

```diff
@@ -34,24 +24,21 @@ public static void Run()
                 {
                     if (WaitForMutex(mutex))
                     {
-                        if (!File.Exists(expectedMigrationFilename))
+                        try
                         {
                             // Only run migrations that have not already been run
-                            int highestMigrationRun = GetHighestMigrationRun(migrationsDirectory);
-                            for (int i = highestMigrationRun + 1; i < Migrations.Count; i++)
+                            if (!File.Exists(expectedMigrationFilename))
                             {
-                                try
-                                {
-                                    Migrations[i]();
-                                    // Create file for every migration run, so that if an older version of NuGet is run, it doesn't try to run
-                                    // migrations again.
-                                    string migrationFile = Path.Combine(migrationsDirectory, (i + 1).ToString(CultureInfo.InvariantCulture));
-                                    File.WriteAllText(migrationFile, string.Empty);
-                                }
-                                catch { }
+                                Migration1.Run();
+                                // Create file for the migration run, so that if an older version of NuGet is run, it doesn't try to run migrations again.
+                                File.Create(expectedMigrationFilename).Dispose();
```

Comment:
Thinking about this more, I actually withdraw my File.Create.Dispose suggestion.  File.WriteAllText can actually do better (and does now in .NET 7) as it doesn't actually need to materialize the stream.  Thanks for taking the feedback, but I'd suggest now putting it back to `File.WriteAllText(migrationFile, string.Empty)` as you previously had it.